### DISCUSSION TITLE
[Feature Proposal] Add minification and purgecss options to CLI

### DISCUSSION THREAD

**Discussion by MattStypa** [tailwindlabs/tailwindcss#546] [Ideas]
# Summery

This addresses a pain point of using Tailwinds with Create-React-App where the Webpack build process is concealed, or in small projects without build process.

# Example

```
./node_modules/.bin/tailwind build ./src/styles.css --minify --purge ./src/static/**/*.html --purge ./src/views/**/*.vue
```

# Implementation

Two new options will be added to the build command.

`-m` `--minify` - Performs minification via `clean-css`
`-p [glob]` `--purge [glob]` - Removes unused css via `purgecss`. Allows for multiple globs to be specified

**Comment by tehpsalmist** [comment]
Do you intend to make purgeCSS a dependency of tailwind in order to make this possible? Or is this something that can work or be ignored based on whether the user has already installed purgecss in their project?

**Comment by MattStypa** [comment]
PurgeCSS would be a new dependency.

**Comment by tehpsalmist** [comment]
I did something similar with my ng-tailwindcss package. https://github.com/tehpsalmist/ng-tailwindcss/blob/master/lib/purge.js

**Comment by brookback** [comment]
I *don't* think this should live inside Tailwind. It should do one thing – not minify and purge. There are other processing tools for that.

**Comment by Ore4444** [comment]
yeah, having PostCSS already in your toolchain, running tailwind for you, why would you want this css utility library to post-process your css?

**Comment by n1ghtmare** [comment]
Since PurgeCSS was added as a dependency and the new tailwind config has a `purge` option, have you considered adding minification as well?

**Comment by LZL0** [comment]
Bump.

**Comment by mike-seekwell** [comment]
I wouldn't add minify when there's tools like like `cleancss`. Maybe adding it to the docs could be helpful.

https://github.com/jakubpawlowicz/clean-css-cli

```
cleancss -o tailwind.min.css tailwind.css
```
